keps/sig-scheduling/4815-dra-partitionable-devices/README.md

```diff
@@ -252,222 +252,470 @@ non-goals of this KEP.
-well as track any capacities consumed from one device by another through its
-`ConsumesCapacityFrom` field. More details on the actual algorithm the
-scheduler follows to make allocation decisions based on the
-`ConsumesCapacityFrom` field can be found in the Design Details section below.
+"flatten" all devices and capacity pools by applying any referenced mixins as
+well as track the capacities consumed by allocated devices. More details
+on the actual algorithm the scheduler follows to make allocation decisions
+based on the capacity pools can be found in the Design Details section below.
 
 ## Design Details
 
 The exact set of proposed API changes can be seen below:
 ```go
 // ResourceSliceSpec contains the information published by the driver in one ResourceSlice.
 type ResourceSliceSpec struct {
-	...
-
-	// DeviceMixins represents a list of device mixins, i.e. a collection of
-	// shared attributes and capacities that an actual device can "include"
-	// to extend the set of attributes and capacities it already defines.
-	//
-	// The main purposes of these mixins is to reduce the memory footprint
-	// of devices since they can reference the mixins provided here rather
-	// than duplicate them.
-	//
-	// The total number of mixins, basic devices, and composite devices must be
-	// less than 128.
-	//
-	// +optional
-	// +listType=atomic
-	DeviceMixins []DeviceMixin `json:"deviceMixins,omitempty"`
+  ...
+
+  // DeviceMixins represents a list of device mixins, i.e. a collection of
+  // shared attributes and capacities that an actual device can "include"
+  // to extend the set of attributes and capacities it already defines.
+  //
+  // The main purposes of these mixins is to reduce the memory footprint
+  // of devices since they can reference the mixins provided here rather
+  // than duplicate them.
+  //
+  // The total number of device mixins, capacity pool mixins, basic devices,
+  // and composite devices must be less than 128.
+  //
+  // +optional
+  // +listType=atomic
+  DeviceMixins []DeviceMixin
+
+  // CapacityPoolMixins represents a list of capacity pool mixins, i.e.
+  // a collection of capacities that a CapacityPool can "include"
+  // to extend the set of capacities it already defines.
+  //
+  // The main purposes of these mixins is to reduce the memory footprint
+  // of capacity pools since they can reference the mixins provided here rather
+  // than duplicate them.
+  //
+  // The total number of device mixins, capacity pool mixins, basic devices,
+  // and composite devices must be less than 128.
+  //
+  // +optional
+  // +listType=atomic
+  CapacityPoolMixins []CapacityPoolMixin
+
+  // SharedCapacityPools defines a list of capacity pools, each of which
+  // has a name and a list of capacities available in the pool.
+  //
+  // The names of the pools must be unique in the ResourceSlice.
+  //
+  // The maximum number of pools is 32.
+  //
+  // +optional
+  // listType=atomic
+  SharedCapacityPools []CapacityPool
```

Comment:
I'm not really sure if the `Shared` prefix here adds any value. I think maybe we should just remove it.

Reply:
Yeah, it's iffy. But I want to differentiate between this and [per-device allocatable resources](https://github.com/kubernetes/enhancements/pull/5104#discussion_r1943844212), which are for user consumption.

Reply:
I did remove the `Shared` prefix in the updated design. I think it works pretty well without it, but let me know if you disagree.